### ISSUE TITLE
Gzip-compressed kernel modules

### ISSUE THREAD

**Issue by travankor** [anatol/booster#55]
This happens on both universal and host image creation. Image is created, but it's hidden with `.` prefix.

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x4c5a1d]

goroutine 913 [running]:
io/ioutil.readAll.func1(0xc0027deea0)
        /usr/lib/go/src/io/ioutil/ioutil.go:30 +0x106
panic(0x5d26e0, 0x7208b0)
        /usr/lib/go/src/runtime/panic.go:969 +0x1b9
bytes.(*Buffer).ReadFrom(0xc00291a628, 0x0, 0x0, 0x0, 0x0, 0xc00291a660)
        /usr/lib/go/src/bytes/buffer.go:204 +0x7d
io/ioutil.readAll(0x0, 0x0, 0x200, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/lib/go/src/io/ioutil/ioutil.go:36 +0xe5
io/ioutil.ReadAll(...)
        /usr/lib/go/src/io/ioutil/ioutil.go:45
main.(*Kmod).addModulesToImage.func1(0xc002f58533, 0x3)
        /builddir/booster-0.3/_build-booster-xbps/src/github.com/anatol/booster/generator/kmod.go:257 +0x4bb
created by main.(*Kmod).addModulesToImage
        /builddir/booster-0.3/_build-booster-xbps/src/github.com/anatol/booster/generator/kmod.go:276 +0x19b
```

Full debug log here: http://ix.io/2SUq
List of kernel modules: http://ix.io/2SUr

**Comment by anatol** [comment]
This nil pointer dereference is probably comes from the fact that booster handles modules with ".ko" , ".xz", ".zst" extensions only:

```go
switch path.Ext(p) {
case ".ko":
        r = f
case ".xz":
        r, err = xz.NewReader(f, 0)
case ".zst":
        r, err = zstd.NewReader(f)
}
content, err := ioutil.ReadAll(r)
```

So it looks like you have a kernel module compressed with some other extension. What compression do you use for modules? I am happy to add its decompression here.


Note that `master` branch improves error handling here and adds

```go
default:
    err = fmt.Errorf("unknown module compression format: %s", ext)
```

so a user will get a more readable error message.

**Comment by travankor** [comment]
The extension is `.ko.gz`, which is gzip. Should hopefully be easy to implement.

**Comment by anatol** [comment]
`*.gz` and `*.lz4` kernel modules support added to `master`. Please take a look.

My understanding is that you package booster for Void Linux. I suggest to use `master` for now. Once all the issue you see resolved I start preparing for `0.4` release.
